app/services/service_provider_config.rb

```diff
@@ -0,0 +1,22 @@
+class ServiceProviderConfig
+  def initialize(filename:, issuer:)
```

Comment:
That is Ruby 2.0's keyword arguments. The colon means the arguments are required, and to use this method, you have to name the arguments, which makes it easy to understand what each arg represents. For example, instead of calling a method like this:
```ruby
method_with_vague_args('service_providers.yml', 'foo')
```
you do this:
```ruby
method_with_clear_args(filename: 'service_providers.yml', issuer: 'test.host')
```
More info here:
https://robots.thoughtbot.com/ruby-2-keyword-arguments